### PR TITLE
Bugfix 1238623- Allow calling pushlog_client functions with tipsonly=False

### PR DESCRIPTION
**Pull request by MikeLing**

### CODE REVIEW

**Comment by MikeLing** [comment]
Hi @armenzg , I consider about it again and think maybe we don't need to indicate in code that only the tip changeset have been loaded. Hmm, in fact I have a little confuse about the meaning about it.

First of all, I think the way we define changsets list in Push object is no matter how many changsets it has, we just load them into Push object. If we only load tip most changeset into Push object, it doesn't make difference to us. On the other hand, we haven't give users too much options anyway. ;) As you mentioned in the bug description, we hard code it in a lot of places (e.g https://hg.mozilla.org/hgcustom/version-control-tools/file/default/pylib/mozhginfo/mozhginfo/pushlog_client.py#l76)

We just need to point that out in the document and tell people we usually return the tip most changeset to them. And for query_pushes_by_revision_range function, people can choose load all changesets with tipsonly=False.

Please tell me if I have any misunderstanding for this, thank you :)

**Comment by MikeLing** [comment]
Oh, BTW, for add exact_revision_from_push function(https://github.com/mozilla/mozilla_ci_tools/pull/409#discussion-diff-49367512R282). I think we could add a option for all functions like:
```
query_pushes_by_pushid_range(......., string=True):
......

      if string:
           return exact_revision_from_push(pushes)
......
```

File: pylib/mozhginfo/mozhginfo/pushlog_client.py [hunk 1]
```diff
@@ -41,13 +42,21 @@ def query_pushes_by_revision_range(repo_url, from_revision, to_revision, version
...
+            version,
+            1
+        )
+    else:
+        url = "%s?fromchange=%s&tochange=%s&version=%d" % (
+            JSON_PUSHES % {"repo_url": repo_url},
+            from_revision,
+            to_revision,
+            version
+        )
```
Comment (armenzg) [inline pylib/mozhginfo/mozhginfo/pushlog_client.py]: I would do:
url = ...
if tipsonly:
    url += '&tipsonly=1'

**Comment by armenzg** [comment]
I like your suggestion on https://github.com/mozilla/version-control-tools/pull/6#issuecomment-171648643
I think it is a good idea. Thanks @MikeLing !

File: pylib/mozhginfo/mozhginfo/pushlog_client.py [hunk 2]
```diff
@@ -41,13 +51,16 @@ def query_pushes_by_revision_range(repo_url, from_revision, to_revision, version
     version       - version of json-pushes to use (see docs)
```
Comment (armenzg) [inline pylib/mozhginfo/mozhginfo/pushlog_client.py]: Could you please add the string doc for the new variable string?

File: pylib/mozhginfo/mozhginfo/pushlog_client.py [hunk 3]
```diff
@@ -25,13 +25,23 @@
...
 VALID_CACHE = {}
+VERSION = 2
 
 
 class PushlogError(Exception):
     pass
 
 
-def query_pushes_by_revision_range(repo_url, from_revision, to_revision, version=2, tipsonly=1):
+def exact_revision_from_push(pushes):
```
Comment (armenzg) [inline pylib/mozhginfo/mozhginfo/pushlog_client.py]: Could you please call it _pushes_to_list()?

File: pylib/mozhginfo/mozhginfo/pushlog_client.py [hunk 4]
```diff
@@ -25,13 +25,23 @@
...
+    """accept a list of push object and return a list of revision"""
```
Comment (armenzg) [inline pylib/mozhginfo/mozhginfo/pushlog_client.py]: objects <- plural
revisions <- plural

File: pylib/mozhginfo/mozhginfo/pushlog_client.py [hunk 5]
```diff
@@ -41,13 +51,16 @@ def query_pushes_by_revision_range(repo_url, from_revision, to_revision, version
...
         JSON_PUSHES % {"repo_url": repo_url},
         from_revision,
         to_revision,
-        version,
-        tipsonly
+        version
     )
+
+    if tipsonly:
+        url += '&tipsonly=1'
```
Comment (armenzg) [inline pylib/mozhginfo/mozhginfo/pushlog_client.py]: Great!

File: pylib/mozhginfo/mozhginfo/pushlog_client.py [hunk 6]
```diff
@@ -25,13 +25,23 @@
...
+def exact_revision_from_push(pushes):
+    """accept a list of push object and return a list of revision"""
+    revisions = []
+    for push in pushes:
+        revisions = revisions + [changeset.node for changeset in push.changesets]
+    return revisions
+
+
+def query_pushes_by_revision_range(repo_url, from_revision, to_revision, version=VERSION,
+                                   tipsonly=True, string=False):
```
Comment (armenzg) [inline pylib/mozhginfo/mozhginfo/pushlog_client.py]: Could you please change the variable 'string' to 'return_revision_list'? That way it holds the purpose in the variable name.

File: pylib/mozhginfo/mozhginfo/pushlog_client.py (hunk 3 above)
Comment (armenzg) [inline pylib/mozhginfo/mozhginfo/pushlog_client.py]: The same with the remaining instances.

**Comment by armenzg** [comment]
Well done. This is now a matter of very minor adjustments.
Once you address it please let me know and I will land it on Mercurial.

Thanks @MikeLing !

**Comment by MikeLing** [comment]
Sure! 30 minutes is all I need to make it happen :)

File: pylib/mozhginfo/mozhginfo/pushlog_client.py (hunk 6 above)
Comment (MikeLing) [inline pylib/mozhginfo/mozhginfo/pushlog_client.py]: Hmm, I just want to make it more simpler. How about something like "meta_string" or "revisions_only"?

Comment (armenzg) [inline pylib/mozhginfo/mozhginfo/pushlog_client.py]: It's hard to convey the meaning with a shorter variable name.
'revision_list' could still work. Is that good for you?

**Comment by armenzg** [comment]
Could you please remove this PR? thanks!

**Pull request closed without merging**